### PR TITLE
feat!: add optional properties to parameterized jobs

### PR DESCRIPTION
**Pull request by Jaryt**

### CODE REVIEW

**Review by brivu — APPROVED** [review]
LGTM!

File: src/lib/Components/Job/index.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ export class Job implements Generable, Executable {
     name: string,
     executor: AnyExecutor,
     steps: Command[] = [],
-    properties?: JobExtraProperties,
+    properties?: JobOptionalProperties,
```
Comment (tapesec) [inline src/lib/Components/Job/index.ts:51]: The JS doc needs to be updated with @param properties

**Review by tapesec — COMMENTED** [review]
Thank you 👍

**Comment by mathieutu** [comment]
We also have to add it in JobContentsShape 🙂.

**Pull request merged**